### DISCUSSION TITLE
How to get NextJS Image component to work with Framer Motion

### DISCUSSION THREAD

**Discussion by lawebb-dev** [vercel/next.js#37001] [Help] [answered]
Hello, I am trying to get framer motion to work with the NextJS image component. I have read online that animating a div and putting the image inside it will work, however this doesn't seem to work, because the tags within the Image component itself are not motion tags. Is there anyway to access the NextJS/Image component to make the span a motion.span and the img a motion.img tag? Or to make it work with Framer Motion in any other way?

**Comment by icyJoseph** [comment, marked as answer]
Mmm if wrapping `next/image` on a `motion.div` doesn't cut it, then I am not sure how else could this be done. 


A solution could've been the snippet under, but the next/image component doesn't forward refs. Taken from [the framer motion docs](https://www.framer.com/docs/component/#custom-components).

```jsx
// ❌  Doesn't work because the next/image component doesn't forward refs
const ExoticImage = forwardRef<HTMLImageElement, ImageProps>(
  function ExoticImageWrapper(props, ref) {
    return <Image  {...props} ref={ref} />; // ❌  this is not forwarded
  }
);

const MotionComponent = motion(ExoticImage)
```

It is a bit unfortunate, because it was closed without resolution, https://github.com/vercel/next.js/issues/18398, and here's clearly a use case. However, I think that wrapping the Image with a div should be enough, not sure what's the issue there.

> because the tags within the Image component itself are not motion tags

Do you have an example? Reproduction demo?

**Reply by lawebb-dev** [reply]
@icyJoseph I have actually got it to work now. The previous issue was with my quite complicated absolute/relative positioning arrangements in the area and then with my setting of some div heights. However I did look into the ref passing solution when I was trying to figure it out and I agree that could help, so will mark you as correct.

**Reply by kschmott** [reply]
What solution did you use to solve this issue?

**Reply by TCNOco** [reply]
We will never know

**Reply by schroeder-g** [reply]
> A solution could've been the snippet under, but the next/image component doesn't forward refs. Taken from [the framer motion docs](https://www.framer.com/docs/component/#custom-components).
> 
> ```js
> // ❌  Doesn't work because the next/image component doesn't forward refs
> const ExoticImage = forwardRef<HTMLImageElement, ImageProps>(
>   function ExoticImageWrapper(props, ref) {
>     return <Image  {...props} ref={ref} />; // ❌  this is not forwarded
>   }
> );
> 
> const MotionComponent = motion(ExoticImage)
> ```


As of Next version `13.06` @icyJoseph 's proposal ought to work, as the `<Image>` component now accepts a ref.

**Comment by lawebb-dev** [comment]
You don't really need to do anything special anymore because NextJS is
removing the span wrapper on its image component. That's what was causing
the issue before. I can't actually remember what I did to resolve the
issue. I did find a solution but it did still make some animations simply
impossible.
